osidb/api_views.py

```diff
@@ -345,6 +354,75 @@ def create(self, request, *args, **kwargs):
         return response
 
 
+@include_meta_attr_extend_schema_view
+@include_exclude_fields_extend_schema_view
+@extend_schema_view(
+    create=extend_schema(
+        request=FlawAcknowledgmentPostSerializer,
+    ),
+    update=extend_schema(
+        request=FlawAcknowledgmentPutSerializer,
+    ),
+)
+class FlawAcknowledgmentView(ModelViewSet):
+    serializer_class = FlawAcknowledgmentSerializer
+    # As long as SRTNotesBuilder.generate_acknowledgments() builds from FlawMeta,
+    # POST, DELETE and PUT don't work.
+    # TODO: Resolve in a follow-up to PR 265.
+    http_method_names = get_valid_http_methods(
+        ModelViewSet, excluded=["post", "delete", "put"]
+    )
+    permission_classes = [IsAuthenticatedOrReadOnly]
+    filterset_class = FlawAcknowledgmentFilter
+
+    def destroy(self, request, *args, **kwargs):
+        bz_api_key = request.META.get("HTTP_BUGZILLA_API_KEY")
+        if not bz_api_key:
+            raise ValidationError({"Bugzilla-Api-Key": "This HTTP header is required."})
+        instance = self.get_object()
+        self.perform_destroy(instance, bz_api_key=bz_api_key)
+        return Response(status=HTTP_200_OK)
+
+    def perform_destroy(self, instance, bz_api_key):
+        """
+        override the default behavior to proxy the delete to Bugzilla
+        """
+        flaw = instance.flaw
+        instance.delete()
+        flaw.save(bz_api_key=bz_api_key)
+
+    def get_flaw(self):
+        """
+        Get flaw object instance.
+        Similar to FlawView.get_object().
+        """
+        pk = self.kwargs["flaw_id"]
+        if CVE_RE_STR.match(pk):
+            obj = get_object_or_404(Flaw, cve_id=pk)
+        else:
+            obj = get_object_or_404(Flaw, uuid=pk)
+        self.check_object_permissions(self.request, obj)
+        return obj
+
+    def get_queryset(self):
+        if getattr(self, "swagger_fake_view", False):
```

Comment:
Remove this condition because we use `drf-spectacular` and not `drf-yasg`.

Reply:
This is actually a common problem to [both libraries](https://drf-spectacular.readthedocs.io/en/latest/faq.html#my-get-queryset-depends-on-some-attributes-not-available-at-schema-generation-time) and it looks like `drf-spectacular` might've been inspired by `drf-yasg` and I assume the reason for it being needed here is because of the nesting.

Reply:
Thanks for the clarification. In that case, we should keep that condition with the link you mentioned to explain why it is important.

Reply:
Will fix the comment.

The schema can be generated without this condition, but it generates a warning.

Reply:
Fixed in commits 09866d24b5b72af9f19ea30c518961a04686bd1b and eaaa28d2c2b1ea423c782b4745faa28b97bf3e7d